src/Fhir.Anonymizer.Tool/FhirResourceDataProcessor.cs

```diff
@@ -74,33 +77,32 @@ public void AnonymizeBulkDataFolder(string inputFolder, string outputFolder, boo
                     Directory.CreateDirectory(resourceOutputFolder);
                 }
 
-                var processedCount = 0;
-                var processedErrorCount = 0;
+                int completedCount = 0;
+                int failedCount = 0;
                 using (FileStream inputStream = new FileStream(bulkResourceFileName, FileMode.Open))
                 using (FileStream outputStream = new FileStream(bulkResourceOutputFileName, FileMode.Create))
                 {
-                    using StreamReader reader = new StreamReader(inputStream);
-                    using StreamWriter writer = new StreamWriter(outputStream);
+                    using FhirStreamReader reader = new FhirStreamReader(inputStream);
+                    using FhirStreamConsumer consumer = new FhirStreamConsumer(outputStream);
+                    Func<string, string> anonymizeFunction = (content) => _engine.AnonymizeJson(content);
+
+                    Stopwatch stopWatch = new Stopwatch();
+                    stopWatch.Start();
 
-                    string resourceLine;
-                    string resultLine;
-                    while ((resourceLine = reader.ReadLine()) != null)
+                    FhirPartitionedExecutor executor = new FhirPartitionedExecutor(reader, consumer, anonymizeFunction);
+                    executor.PartitionCount = Environment.ProcessorCount;
+                    Progress<BatchAnonymizeProgressDetail> progress = new Progress<BatchAnonymizeProgressDetail>();
+                    progress.ProgressChanged += (obj, args) =>
                     {
-                        try
-                        {
-                            resultLine = _engine.AnonymizeJson(resourceLine);
-                            writer.WriteLine(resultLine);
-                            processedCount += 1;
-                        }
-                        catch (Exception innerException)
-                        {
-                            processedErrorCount += 1;
-                            Console.Error.WriteLine($"Error #{processedErrorCount}\nResource: {resourceLine}\nErrorMessage: {innerException.ToString()}");
-                        }
-                    }
+                        Interlocked.Add(ref completedCount, args.Completed);
+                        Interlocked.Add(ref failedCount, args.Failed);
+                        Console.WriteLine($"[{stopWatch.Elapsed.ToString()}]: {completedCount} Completed. {failedCount} Failed.");
+                    };
+
+                    executor.ExecuteAsync(CancellationToken.None, false, progress).Wait();
```

Comment:
Anonymize (CPU intensive) is parallelized here, but file writing is not. Does it make sense to spawn multiple executors in parallel each working on separate file (assuming many files in bulk data folder)?